### ISSUE TITLE
Text mode file operations

### ISSUE THREAD

**Issue by sjakobi** [fpco/unliftio#51]
`UnliftIO.IO.File` currently offers only binary mode file operations.

It would be nice if text mode operations were offered too, to simplify writing text files in a cross-platform way: On Windows, text mode writes change `LF` line endings to `CRLF`. See [`openBinaryFile`](http://hackage.haskell.org/package/base-4.12.0.0/docs/System-IO.html#v:openBinaryFile):

> On Windows, reading a file in text mode (which is the default) will translate CRLF to LF, and writing will translate LF to CRLF. This is usually what you want with text files. With binary files this is undesirable; also, as usual under Microsoft operating systems, text mode treats control-Z as EOF. Binary mode turns off all special treatment of end-of-line and end-of-file characters. (See also hSetBinaryMode.)

**Comment by sjakobi** [comment]
Hmm, if I understand `hSetBinaryMode` correctly, I can already use it to switch to text mode with `hSetBinaryMode False`.

So `writeFileAtomic` could be implemented as

```
writeFileAtomic fp bytes =
  liftIO $ withBinaryFileAtomic fp WriteMode $ \h -> do
    hSetBinaryMode h False
    B.hPut h bytes
```

**Comment by snoyberg** [comment]
Yes, that would be my recommended approach. I don't have a strong opinion, but I'm weakly against adding more helper functions to the API due to expanding things too much.
